test/manipulation.test.js

```diff
@@ -1097,6 +1098,53 @@ describe('manipulation', function() {
     });
   }
 
+  var hasReplaceById = !!getSchema().connector.replaceById;
+  bdd.describeIf(hasReplaceById, 'replaceOrCreate when forceId is true', function() {
+    var Post, foundId;
+    var ds = getSchema();
```

Comment:
I think we should initialise `ds` inside a `before` hook (i.e. just before we run the tests), not at the time when `describe` is invoked. My main concern is that if `getSchema` is statefull, then the `ds` instance may become invalid by the time the tests are run, because other `before` hooks are calling `getSchema` in the meantime.

Also if you tell mocha to run a subset of tests, e.g. via `mocha -g <pattern>` or by using `it.only/describe.only`, then your current code will always create the datasource, even though "replaceOrCreate" tests are not run at all. When `getSchema()` is moved to a `before` hook, it won't be run if "replaceOrCreate" tests are skipped.